gtsfm/loader/loader_base.py

```diff
@@ -10,15 +10,30 @@
 from dask.delayed import Delayed
 from gtsam import Cal3Bundler, Pose3
 
+import gtsfm.utils.images as img_utils
+import gtsfm.utils.logger as logger_utils
 from gtsfm.common.image import Image
 
 
+logger = logger_utils.get_logger()
+
+
 class LoaderBase(metaclass=abc.ABCMeta):
```

Comment:
I think the base class could do more. Are there or will there be any loaders that do not pass a path to a directory of images to be loaded? It looks like all loaders are currently doing the same thing.

Reply:
This is true, they are all pretty similar -- not in the constructor, but yes for intrinsics, extrinsics, fetching images. Definitely something for us to consider unifying in the future. Maybe get_image() should be identical for all, and so should get_full_res_image(). And then if someone wants to have their own function in the future, they can override the parent method?

Reply:
We would need to force every user to define an `image_paths` list though, and a wTi_list, etc. I think each dataset might have some different assumptions so probably would want to allow overriding the methods if needed when fetching particular data.

Reply:
Yeah overriding is fine, though currently not used.

Reply:
I don't think forcing every user to define an `image_paths` is very restrictive